### PR TITLE
Update ompl_interface_tutorial.rst

### PR DESCRIPTION
**Pull request by ejalaa12**
Update the tutorial to explain how to create a Custom Optimization Objective for OMPL as described in this pull request #1436 (https://github.com/ros-planning/moveit2/pull/1436)

### CODE REVIEW

File: doc/examples/ompl_interface/ompl_interface_tutorial.rst [hunk 1]
```diff
@@ -99,7 +99,32 @@ The configuration of these optimization objectives can be done in the *ompl_plan
         goal_bias: 0.05
         delay_collision_checking: 1
 
-Other optimization objectives can be defined programmatically. For more information on the OMPL optimal planners, the reader is referred to the `OMPL - Optimal Planning documentation <http://ompl.kavrakilab.org/optimalPlanning.html>`_.
+Other optimization objectives can be defined programmatically and then passed to MoveIt.
+
+To do so, a minimal setup is as follows :
+
+#. In your own package, create your an optimization objective class for OMPL. For more information on the OMPL optimal planners, the reader is referred to the `OMPL - Optimal Planning documentation <http://ompl.kavrakilab.org/optimalPlanning.html>`_.
```
Comment (mamoll) [inline doc/examples/ompl_interface/ompl_interface_tutorial.rst:106]:
```suggestion
#. In your own package, create your own optimization objective class for OMPL. For more information on the OMPL optimal planners, the reader is referred to the `OMPL - Optimal Planning documentation <http://ompl.kavrakilab.org/optimalPlanning.html>`_.
```

File: doc/examples/ompl_interface/ompl_interface_tutorial.rst [hunk 2]
```diff
@@ -99,7 +99,32 @@ The configuration of these optimization objectives can be done in the *ompl_plan
...
+
+#. In your .cpp file, include ``"moveit/ompl_interface/ompl_optimization_objective_loader.h"`` and add, at the end, the macro ``MOVEIT_OPTIMIZATION_OBJECTIVE_PLUGIN(namespace, class_name)``, with the right namespace ans class name you used.
```
Comment (mamoll) [inline doc/examples/ompl_interface/ompl_interface_tutorial.rst:108]:
```suggestion
#. In your .cpp file, include ``"moveit/ompl_interface/ompl_optimization_objective_loader.h"`` and add, at the end, the macro ``MOVEIT_OPTIMIZATION_OBJECTIVE_PLUGIN(namespace, class_name)``, with the right namespace and class name you used.
```

File: doc/examples/ompl_interface/ompl_interface_tutorial.rst [hunk 3]
```diff
@@ -99,7 +99,32 @@ The configuration of these optimization objectives can be done in the *ompl_plan
...
+
+#. Create a plugin description file .xml containing the following : ::
+
+    <library path="package_name">
+    <class name="custom_optimizer"
```
Comment (mamoll) [inline doc/examples/ompl_interface/ompl_interface_tutorial.rst:113]:
```suggestion
    <class name="custom_objective"
```

File: doc/examples/ompl_interface/ompl_interface_tutorial.rst [hunk 4]
```diff
@@ -99,7 +99,32 @@ The configuration of these optimization objectives can be done in the *ompl_plan
...
+           base_class_type="ompl_optimization_loader::OptimizationObjectiveLoader">
+        <description> </description>
+    </class>
+    </library>
+
+   The ``name`` tag will the name to pass the parameter **optimization_objective**.
+
+   The ``type`` tag is the fully qualified type of the plugin, be sure to add ``Loader`` at the end of the tag type.
+
+   The ``bass_class`` is the fully qualified base class type for the plugin and does not need to be modified.
```
Comment (mamoll) [inline doc/examples/ompl_interface/ompl_interface_tutorial.rst:124]:
```suggestion
   The ``bass_class_type`` is the fully qualified base class type for the plugin and does not need to be modified.
```
Or change `base_class_type` in the XML snippet above to `base_class`?

**Review by mamoll — CHANGES_REQUESTED** [review]

**Comment by ejalaa12** [comment]
@mamoll Thanks for the review, just fixed all the typos :)

File: doc/examples/ompl_interface/ompl_interface_tutorial.rst [hunk 5]
```diff
@@ -99,7 +99,32 @@ The configuration of these optimization objectives can be done in the *ompl_plan
...
+           base_class_type="ompl_optimization_loader::OptimizationObjectiveLoader">
+        <description> </description>
+    </class>
+    </library>
+
+   The ``name`` tag will the name to pass the parameter **optimization_objective**.
+
+   The ``type`` tag is the fully qualified type of the plugin, be sure to add ``Loader`` at the end of the tag type.
+
+   The ``bass_class_type`` is the fully qualified base class type for the plugin and does not need to be modified.
```
Comment (stephanie-eng) [inline doc/examples/ompl_interface/ompl_interface_tutorial.rst:124]:
```suggestion
   The ``base_class_type`` is the fully qualified base class type for the plugin and does not need to be modified.
```

**Comment by ejalaa12** [comment]
@stephanie-eng Friendly ping :)

**Review by sea-bass — APPROVED** [review]
I'm not sure why this fell through, but I was scrolling and it looks good to me.

**Comment by stephanie-eng** [comment]
I thought this is based on https://github.com/moveit/moveit2/pull/1436 which is not merged yet?

**Comment by sea-bass** [comment]
Oh no. I did not realize it was documenting a new feature. I'll revert.

**Comment by sea-bass** [comment]
Reverted in https://github.com/moveit/moveit2_tutorials/commit/513ce271d41fb9424543a67b689c70f497d37856

If this feature lands at some point, please feel free to open up a new pull request here.

Thanks @stephanie-eng!

**Pull request merged**